daemon/container_operations_unix.go

```diff
@@ -344,6 +345,16 @@ func killProcessDirectly(cntr *container.Container) error {
 	if status.Err() != nil {
 		// Ensure that we don't kill ourselves
 		if pid := cntr.GetPID(); pid != 0 {
+
+			// since we can not kill a zombie pid, add zombie check here
+			isZombie, err := system.IsProcessZombie(pid)
+			if err != nil {
+				logrus.Warningf("Container %s state is invalid", stringid.TruncateID(cntr.ID))
+				return err
+			}
+			if isZombie {
+				return fmt.Errorf("container %s PID %d is zombie, can not be killed", stringid.TruncateID(cntr.ID), pid)
```

Comment:
Perhaps we should include some information about recommending `--init` for containers to prevent this situation

Reply:
done